### ISSUE TITLE
[web]: Cannot call rest api from a self signed https server

### ISSUE THREAD

**Issue by Stemicio9** [flutter/flutter#51146]
Flutter 1.12.13+hotfix.6 
Dart 2.7.0

I am looking for a way to trust my self signed certificate in a rest call. For NON web applications it could be done with HttpClient (dart:io), but if I use it in web I have an exception. Libraries http, http_client and dio cannot face this issue (dio has something, but uses HttpClient so it throws the same exception). 
Is it considerable as an issue?

**Comment by iapicca** [comment]
Hi @Stemicio9 
can you please provide your `flutter run -d chrome --verbose` 
showing the exception you are mentioning
and if possible a reproducible minimal code sample.
Thank you

**Comment by Stemicio9** [comment]
Ok
I have a simple login form that calls a function to call a post request (a simple oauth2 login request) to a spring boot application hosted into my local network. 

Here a snippet of my code: 
`import 'dart:io';
import 'package:http/io_client.dart';
import 'package:w1nbackofficedart/login/loginpag.dart';
import 'package:http/http.dart';

final baseurl = "https://myipaddress:8443/";

 Future<bool> entra(username, password) async {
    final authorizationEndpoint =
    Uri.parse(baseurl + "oauth/token");
    final identifier = "myidentifier";
    final secret = "mysecret";
    var body = {
      "grant_type": "password",
      "username": username,
      "password": password
    };
    var headers = <String, String>{};
    headers['Authorization'] = basicAuthHeader(identifier, secret);
    HttpClient client = new HttpClient();
    // To trust every CA
    client.badCertificateCallback = (a,b,c){return true;};

    IOClient ioClient = new IOClient(client);
    var response = await ioClient.post(authorizationEndpoint.toString(), body: body,headers: headers);
    print(response.body);
  }
`

Here the error : 
`Uncaught (in promise) Error: Unsupported operation: Platform._version
    at Object.throw_ [as throw] (errors.dart:196)
    at Function._version (io_patch.dart:286)
    at Function.get version [as version] (platform_impl.dart:123)
    at get _version (platform.dart:76)
    at Function.desc.get [as _version] (utils.dart:77)
    at Function.get version [as version] (platform.dart:233)
    at Object._getHttpVersion (http_impl.dart:3332)
    at new _http._HttpClient.new (http_impl.dart:2146)
    at Function.new (http.dart:1491)
    at entra (loginservice.dart:50)    <------------------------------
    at entra.next (<anonymous>)
    at runBody (async_patch.dart:86)
    at Object._async [as async] (async_patch.dart:125)
    at loginservice.LoginService.new.entra (loginservice.dart:30)
    at loginpag.NuovoLoginState.new.risultatologin (loginpag.dart:122)
    at risultatologin.next (<anonymous>)
    at runBody (async_patch.dart:86)
    at Object._async [as async] (async_patch.dart:125)
    at loginpag.NuovoLoginState.new.risultatologin (loginpag.dart:121)
    at _InkResponseState.new.[_handleTap] (ink_well.dart:706)
    at ink_well.dart:789
    at tap.TapGestureRecognizer.new.invokeCallback (recognizer.dart:182)
    at tap.TapGestureRecognizer.new.handleTapUp (tap.dart:486)
    at tap.TapGestureRecognizer.new.[_checkUp] (tap.dart:264)
    at tap.TapGestureRecognizer.new.handlePrimaryPointer (tap.dart:199)
    at tap.TapGestureRecognizer.new.handleEvent (recognizer.dart:467)
    at pointer_router.PointerRouter.new.[_dispatch] (pointer_router.dart:76)
    at pointer_router.dart:117
    at LinkedMap.new.forEach (linked_hash_map.dart:23)
    at pointer_router.PointerRouter.new.[_dispatchEventToRoutes] (pointer_router.dart:115)
    at pointer_router.PointerRouter.new.route (pointer_router.dart:101)
    at binding$5.WidgetsFlutterBinding.new.handleEvent (binding.dart:218)
    at binding$5.WidgetsFlutterBinding.new.dispatchEvent (binding.dart:198)
    at binding$5.WidgetsFlutterBinding.new.[_handlePointerEvent] (binding.dart:156)
    at binding$5.WidgetsFlutterBinding.new.[_flushPointerEventQueue] (binding.dart:102)
    at binding$5.WidgetsFlutterBinding.new.[_handlePointerDataPacket] (binding.dart:86)
    at _engine.PointerBinding.new.[_onPointerData] (pointer_binding.dart:83)
    at pointer_binding.dart:334
    at HTMLElement.<anonymous> (pointer_binding.dart:183)`


Line 50 into my code (signaled with the arrow in the error) is : 
 HttpClient client = new HttpClient();

Hope it can help, I'll do everything needed.

**Comment by yjbanov** [comment]
Browsers do not allow managing your own certificates. You can use `Client` constructor and it will pick a Web-compatible implementation automatically for you. However, it will use browser's built-in certificates.

Consider posting a StackOverflow question. Perhaps other Web developers already have a solution for this.

**Comment by dhs9004** [comment]
@Stemicio9 Did you find a solution for this? I found a lot of people answer this question but no answer yet.
Thanks,

**Comment by Stemicio9** [comment]
Sincerily not, but I am still working on it

**Comment by its-a-feature** [comment]
Any solution to this yet? I've tried everything I can find online and I still can't get a web instance to connect to a self-signed cert. Either I just always get an error from chrome that I can't do anything about or I run into that same Platform._version bug. I'm using this in a project that allows you to deploy your own web server, so it's absolutely necessary that it works with self-signed certificates. I've tried dio, http, and ioclient stuff. All the same errors
